code/game/objects/items/weapons/grenades/projectile.dm

```diff
@@ -0,0 +1,75 @@
+/obj/item/weapon/grenade/shooter
+	name = "projectile grenade"	// I have no idea what else to call this, but the base type should never be used
+	icon_state = "frggrenade"
+	item_state = "grenade"
+
+	var/list/projectile_types = list(/obj/item/projectile/bullet/pistol/rubber)	// What sorts of projectiles might we make?
+
+	//The radius of the circle used to launch projectiles. Lower values mean less projectiles are used but if set too low gaps may appear in the spread pattern
+	var/spread_range = 7
+
+	loadable = FALSE
+
+/obj/item/weapon/grenade/shooter/detonate()
+	..()
+
+	var/turf/O = get_turf(src)
+	if(!O)
+		return
+
+	src.launch_many_projectiles(O, spread_range, projectile_types)
+
+	qdel(src)
+
+
+
+/obj/item/weapon/grenade/shooter/rubber
+	name = "rubber pellet grenade"
```

Comment:
Just as a note, I usually hear these referred to as 'sting grenades', though that might be based on vernacular and locality.